sklearn/metrics/_dist_metrics.pyx.tp

```diff
@@ -65,20 +65,59 @@ def get_valid_metric_ids(L):
             if (val.__name__ in L) or (val in L)]
 
 cdef class DistanceMetric:
+    """DistanceMetric class
+
+    This class provides a uniform interface to fast distance metric
```

Comment:
Instead of repeating what I would put in the short summary above, I think that we could directly add the text of the "Notes" in the long description.